x-pack/plugins/security_solution/server/lib/telemetry/sender.ts

```diff
@@ -155,12 +160,9 @@ export class TelemetryEventsSender {
   // this returns a cached version of the telemetry opt-in. The cache is updated
   // in the periodically executed sendIfDue function. This function returns false
   // until the first execution of that function.
-  public isTelemetryOptedIn(): boolean {
-    if (!this.isOptedIn) {
-      return false;
-    } else {
-      return this.isOptedIn === true;
-    }
+  public async isTelemetryOptedIn(): Promise<boolean> {
+    this.isOptedIn = await this.telemetryStart?.getIsOptedIn();
```

Comment:
Note that checking for telemetry is not super cheap, because it needs to access a saved object. In this version, we check it twice: once every minute for the local task (`sendIfDue`) and once every 5 minutes in the TaskManager task. It's probably ok like it is, but mentioning only for context on why it was using a cached variable.

Reply:
Yeah, I agree - replied here: https://github.com/elastic/kibana/pull/84422#discussion_r536101223